### PR TITLE
[Feature] Added logging and evaluation functionality

### PR DESCRIPTION
**Pull request by Saf9933**
**_Summary_**
**The two key features:**
   - Added logging to track model loading, configuration, task execution, and evaluation.
   - Added an evaluation function that calculates accuracy after task execution using a test dataset loaded from JSON.
**README Updates:**
   - Updated the README.md file to include a Contributing section.
The section explains how others can fork the repository, create new branches, and submit Pull Requests to contribute to the project.
   
Device Compatibility 
- The code is written to work on systems with or without CUDA support. It will automatically detect if CUDA is available and use it; otherwise, it defaults to the CPU.
Testing issue for my MacBook
Due to hardware limitations (MacBook M1, no CUDA support), I was unable to fully test the code. However, I structured it to run efficiently on a system with CUDA-enabled hardware. I request that someone with the necessary hardware verify the functionality thank you :).

Files Changed:
- mlora_train.py: Added logging, dataset loading, and evaluation.
- README.md: Contributing section

### CODE REVIEW

File: mlora_train.py [hunk 1]
```diff
@@ -20,25 +20,115 @@
...
 import mlora.config
+import logging
+import torch
+from torch.utils.data import DataLoader, Dataset
+import json
+import yaml
+
+
+# Custom dataset class to load data from JSON
+class CustomDataset(Dataset):
```
Comment (yezhengmao1) [inline mlora_train.py:31]: it's better to create a new file and new class .

**Review by yezhengmao1 — CHANGES_REQUESTED** [review]

**Comment by yezhengmao1** [comment]
can we split the function (dataset loading and evaluation) to another file or module?

File: mlora_train.py (hunk 1 above)
Comment (Saf9933) [reply mlora_train.py]: Thank you for the feedback! I’ve now refactored the dataset loading and evaluation into a separate module (dataset_utils.py) as suggested you can check it out!

File: tests/dataset_utils.py
Comment (yezhengmao1) [inline tests/dataset_utils.py:1]: can we move this file into a new directory "dataset" ?

**Review by yezhengmao1 — CHANGES_REQUESTED** [review]
